packages/dev/core/src/contextHelper.ts

```diff
@@ -0,0 +1,90 @@
+declare const require: any;
+
+const ConversionTable: { [key: string]: string } = {
+    vector2: "math.vector",
+    vector3: "math.vector",
+    vector4: "math.vector",
+    matrix: "math.vector",
+    color3: "math.color",
+    color4: "math.color",
+    valuecondition: "condition",
+    predicatecondition: "condition",
+    statecondition: "condition",
+    setparentaction: "directaction",
+    executecodeaction: "directaction",
+    donothingaction: "directaction",
+    stopanimationaction: "directaction",
+    playanimationaction: "directaction",
+    incrementvalueaction: "directaction",
+    setvalueaction: "directaction",
+    setstateaction: "directaction",
+    switchbooleanaction: "directaction",
+    combineaction: "directaction",
+    playsoundaction: "directaudioaction",
+    stopsoundaction: "directaudioaction",
+    followcamera: "followcamera",
+    arcfollowcamera: "followcamera",
+    oppositeblock: "oneminusblock",
+};
+
+const RenameTable: { [key: string]: string } = {
+    oppositeblock: "OneMinusBlock",
+};
+
+/** @internal */
+// eslint-disable-next-line @typescript-eslint/naming-convention
+let _DiscoveredTypes: { [key: string]: any };
+
+function GetLastPartWithoutExtension(path: string) {
+    const parts = path.split("/");
+    const last = parts[parts.length - 1];
+
+    return last.replace(/\.[^/.]+$/, "");
+}
+
+/**
+ * Extract a class from the bundle context
+ * @param name name of the class to extract
+ * @returns the class or undefined if not found
+ */
+export function ExtractClass(name: string) {
+    if (!_DiscoveredTypes) {
+        _DiscoveredTypes = {};
+        // Use require.context to import all modules
+        const requireModule = require.context(

```

Comment:
Is this not webpack specific ? cause this would basically fail for anybody bundling without Webpack ?

Reply:
This is THE question:) hence why I wanted @ryantrem to check

Reply:
I can pull your branch and test it out with the viewer (which uses rollup) to be sure.

Reply:
I commented about it. We can't use require.context if we want to stay esm friendly